Modules/_io/bufferedio.c

```diff
@@ -1559,7 +1560,15 @@ _bufferedreader_raw_read(buffered *self, char *start, Py_ssize_t len)
     do {
         res = PyObject_CallMethodOneArg(self->raw, &_Py_ID(readinto), memobj);
     } while (res == NULL && _PyIO_trap_eintr());
+    PyObject *exc = PyErr_GetRaisedException();
+    release_res = PyObject_CallMethod(memobj, "release", NULL);
+    _PyErr_ChainExceptions1(exc);
     Py_DECREF(memobj);
+    if (release_res == NULL) {
+        Py_XDECREF(res);
+        return -1;
+    }
+    Py_DECREF(release_res);
```

Comment:
This makes sense to me. The performance impact should be minimal as it's only a cleanup method call after the buffer's been entirely read.

However, since `mv.release()` can potentially raise, this is a change in behavior so we have to decide whether the risk of existing code breakage is worth backporting the patch to older releases.

I see @gpshead decided this isn't a security issue so we won't be bringing it to 3.8, 3.9, 3.10. Question is if we want it in 3.11 and 3.12.